### PR TITLE
Only toggle colorlabel if all images have the label

### PR DESCRIPTION
**Pull request by Solarer**
This addresses #10734
Currently setting a colorlabel via F1 inverts the red colorlabel on all images in the selection. 

This can remove the label from images by accident if the user does not realize that he had applied that label before to some of the images.

The code change checks all images for the existence of the label and changes the action from TOGGLE to ADD if at least one image does not have the color label. Therefore color labels will only be removed if all images have the color label.

### CODE REVIEW

File: src/common/colorlabels.c [hunk 1]
```diff
@@ -132,8 +132,29 @@ typedef enum dt_colorlabels_actions_t
 
 
 
-static void _colorlabels_execute(const GList *imgs, const int labels, GList **undo, const gboolean undo_on, const int action)
+static void _colorlabels_execute(const GList *imgs, const int labels, GList **undo, const gboolean undo_on, int action)
 {
+  if(action == DT_CA_TOGGLE)
+  {
+    // if we are supposed to toggle color labels, first check if all images have that label
+    for(const GList *images = imgs; images; images = g_list_next((GList *)images))
```
Comment (TurboGit) [inline src/common/colorlabels.c:140]: Just a nitpicking point, here I would use `GList *image` for the iterator, as it is used to get the image_id for a single image at a time. This is I think the convention used in dt.

Comment (TurboGit) [reply src/common/colorlabels.c:140]: In fact now, the loop just below is using `images`, consistency is hard to keep :) Anyway, feel free to change both loops if you agree.

**Comment by Solarer** [comment]
@TurboGit updated

**Comment by elstoc** [comment]
If you could do the same for the "reject" option, this would also resolve #5138

**Comment by Solarer** [comment]
It also applies to one star ratings. It is on my list

**Review by TurboGit — APPROVED** [review]
Works for me, thanks!

**Pull request merged**